.buildkite/pipeline.ml.yml

```diff
@@ -54,18 +54,18 @@
     - ./ci/env/env_info.sh
     - bazel test --config=ci $(./ci/run/bazel_export_options) --build_tests_only --test_tag_filters=tune,-gpu_only,-ray_air,-gpu,-doctest,-needs_credentials python/ray/train/...
 
-# - label: ":train: :key: Train examples with authentication"
-#   conditions: ["NO_WHEELS_REQUIRED", "RAY_CI_TRAIN_AFFECTED", "RAY_CI_BRANCH_BUILD"]
-#   instance_size: medium
-#   commands:
-#     - if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then exit 0; fi
-#     - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/build/upload_build_info.sh; fi }; trap cleanup EXIT
-#     - TRAIN_TESTING=1 ./ci/env/install-dependencies.sh
-#     - ./ci/env/env_info.sh
-#     - credentials=$(python ./ci/env/setup_credentials.py)
-#     - credential_exit_code=$$?
-#     - if [ $credential_exit_code -eq 0 ]; then echo "Credentials are fetched successfully"; bazel test --config=ci $(./ci/run/bazel_export_options) --build_tests_only --test_tag_filters=needs_credentials 
-#       python/ray/train/... $credentials; else echo "Credentials cannot be fetched"; exit 1; fi
+- label: ":train: :key: Train examples with authentication"
+  conditions: ["NO_WHEELS_REQUIRED", "RAY_CI_TRAIN_AFFECTED", "RAY_CI_BRANCH_BUILD"]
+  instance_size: medium
+  commands:
+    - if [ "$BUILDKITE_PULL_REQUEST" != "false" ]; then exit 0; fi
+    - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/build/upload_build_info.sh; fi }; trap cleanup EXIT
+    - TRAIN_TESTING=1 ./ci/env/install-dependencies.sh
+    - ./ci/env/env_info.sh
+    - python ./ci/env/setup_credentials.py
+    - echo "Credentials are fetched successfully"
+    - bazel test --config=ci $(./ci/run/bazel_export_options) --build_tests_only --test_tag_filters=needs_credentials 
+      python/ray/train/... $(cat /tmp/experiment_tracking_credentials.txt)
```

Comment:
I think this still has the risk that it will print out the credentials into the logs if not very careful.

it is going to be much better if the test itself can read the credentials from secrets manager, and then set it into `os.environ`

Reply:
hmm, this test wants to test some doc code in our documentation. prefer test the doc code as is...